agent_gateway/gateway/gateway.py

```diff
@@ -50,20 +51,20 @@ def __init__(self, session, llm) -> None:
     async def arun(self, prompt: str) -> str:
         """Run the LLM."""
         headers, url, data = self._prepare_llm_request(prompt=prompt)
-        gateway_logger.log(logging.DEBUG, "Cortex Request URL\n", url, block=True)
-        gateway_logger.log(logging.DEBUG, "Cortex Request Data\n", data, block=True)
-
-        response_text = await post_cortex_request(url=url, headers=headers, data=data)
-        gateway_logger.log(
-            logging.DEBUG,
-            "Cortex Request Response\n",
-            response_text,
-            block=True,
-        )
+        # gateway_logger.log(logging.DEBUG, "Cortex LLM Prompt\n", data, block=True)
+
+        try:
+            response_text = await post_cortex_request(
+                url=url, headers=headers, data=data
+            )
+        except Exception as e:
+            raise AgentGatewayError(
+                message=f"Failed Cortex LLM Request. See details:{str(e)}"
+            ) from e
 
         if "choices" not in response_text:
             raise AgentGatewayError(
-                message=f"Failed Cortex LLM Request. Missing choices in response. See details:{response_text}"
+                message=f"Invalid Cortex LLM Response. See details:{response_text}"
```

Comment:
What are the possibilities here?

Would it only hit this block in the case of Analyst not returning either the text response with suggestions or an SQL expression?

Reply:
this is for the complete calls, not analyst. there can be cases where complete is not configured properly (invalid model provided by end user..etc) or the response comes back empty